### PR TITLE
Fix octokit returning weird results from listWorkflowRuns

### PR DESCRIPTION
**Pull request by erd0s**
This is the gist, what you think?

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -73,6 +73,9 @@ async function main() {
                     if (runNumber) {
                         return r.run_number == runNumber
                     }
+                    if (workflowConclusion.length > 0) {
```
Comment (vmirkin-proag) [inline main.js:76]: wrong place to make this check.

File: main.js [hunk 2]
```diff
@@ -73,6 +73,9 @@ async function main() {
...
+                        return workflowConclusion.includes(r.status) || workflowConclusion.includes(r.conclusion)
```
Comment (vmirkin-proag) [inline main.js:77]: wrong place to make the check. if either commit or runNumber is specified this function already returned by now.

File: main.js (hunk 1 above)
Comment (vmirkin-proag) [inline main.js:76]: based on line 16 this check will alwys return true.

**Comment by dawidd6** [comment]
Ok. This is not working as expected and I guess the data returned from GitHub contains wrong statuses of workflow runs but I have to check those yet

**Comment by dawidd6** [comment]
Yep. According to API docs `completed` is automatically assumed when specifying `status`. I think I read that right.

**Comment by erd0s** [comment]
Maybe like this? https://github.com/erd0s/action-download-artifact/commit/9073df9f344ea2c8dc4e8a2d36c4ca81f2b7c654, also taking into account comments from @vmirkin-proag

**Comment by dawidd6** [comment]
Fixed, thanks

**Comment by vnaidin** [comment]
Hello guys, after fix it finds the oldest successful workflow run..., and according to my settings github automatically deletes such old artifacts. Can you have a look again?

**Pull request merged**